### PR TITLE
PR: FE Profile Settings > Profile Photo

### PR DESCRIPTION
**Pull request by ababber**
### What this PR does (required):
-completed ui for uploading and displaying a photo of the user

### Screenshots / Videos (front-end only):
https://user-images.githubusercontent.com/18492009/143304478-83e9d9bb-89f1-43c5-af07-aac736b723b2.mov

### Any information needed to test this feature (required):
- log in, navigate to edit profile page, click on profile photo tab, click on upload photo button, select file from device, and complete process

### Any issues with the current functionality (optional):
- the photo upload is not persistent, this will be implemented during the integration phase

### CODE REVIEW

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Looks nice! Just a few minor comments to fix.

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    </Typography>
+                  </Box>
+                </Box>
+              </Grid>
+            </Grid>
+            <Grid container alignItems="center" direction="column">
+              <Grid item>
+                <Box display="flex" alignItems="center" justifyContent="center">
+                  <Box alignSelf="center">
+                    <input
```
[the remark this answers is not in the document]
Comment (ababber) [reply client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:61]: Please refer to the documentation for an [upload button on the MUI v4 ](https://v4.mui.com/components/buttons/#UploadButtons.tsx). I implemented the upload button just as they suggested.

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                      accept="image/*"
+                      className={classes.input}
+                      id="contained-button-file"
+                      multiple
+                      type="file"
+                      onChange={onPhotoUploadChange}
+                    />
+                    <label htmlFor="contained-button-file">
```
[the remark this answers is not in the document]
Comment (ababber) [reply client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:69]: Please refer to the documentation for an [upload button on the MUI v4 site](https://v4.mui.com/components/buttons/#UploadButtons.tsx). I implemented the upload button just as they suggested.

[the remark this answers is not in the document]
Comment (rajivtitus) [reply client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:76]: @ababber You can substitute the HTML label with a MUI InputLabel 😄

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajivtitus) [reply client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:68]: @ezeilo-su @ababber I think you can stick to using a HTML input here, if I'm not wrong the MUI TextField won't fit our use case here :)

**Review by ezeilo-su — COMMENTED** [review]
Thank you for making those changes :+1:

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+              <Grid item xs={12} sm={8} md={7}>
+                <Box display="flex" alignItems="center" justifyContent="center">
+                  <Box width="100%" maxWidth={450} alignSelf="center">
+                    {!isUploaded ? (
+                      <Avatar className={classes.avatar}>
+                        <AccountCircleIcon className={classes.icon} />
+                      </Avatar>
+                    ) : (
+                      <Avatar className={classes.avatar}>
+                        <img src={displayPhoto} className={classes.photo} />
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:42]: Please include an 'alt' attribute for accessibility purposes :)

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            <Typography color="textPrimary" className={classes.title}>
+              Profile Photo
+            </Typography>
+            <Grid container alignItems="center" direction="column">
+              <Grid item xs={12} sm={8} md={7}>
+                <Box display="flex" alignItems="center" justifyContent="center">
+                  <Box width="100%" maxWidth={450} alignSelf="center">
+                    {!isUploaded ? (
+                      <Avatar className={classes.avatar}>
+                        <AccountCircleIcon className={classes.icon} />
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:38]:
```suggestion
                       {! isUploaded ? <AccountCircleIcon className={classes.icon} /> : <img src={displayPhoto} className={classes.photo} />
```

**Review by rajivtitus — APPROVED** [review]
Great work with the UI 💯 
Just address a couple of minor comments prior to merging, thanks!

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx (hunk 4 above)
Comment (ababber) [reply client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:38]: That definitely simplifies the code. Thanks for the suggestion!

**Pull request merged**